src/DocumentDB.ChangeFeedProcessor/PartitionManagement/ILeaseEx.cs

```diff
@@ -0,0 +1,17 @@
+﻿// ----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+// ----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement
+{
+    /// <summary>
+    /// Provides extensibility for <see cref="ILease"/> interface.
+    /// </summary>
+    public interface ILeaseEx : ILease
+    {
+        /// <summary>
+        /// Gets or sets a reason of acquiring this lease which can be managed from <see cref="IParitionLoadBalancingStrategy"/>.
+        /// </summary>
+        AcquireReason? AcquireReason { get; set; }
```

Comment:
Any specific reason for nullable? Why not just default to default enum value?

Reply:
Acquire reason is not going to be saved to lease collection. Thus this field will be null for all freshly deserialized leases. Do you prefer to have some default value Default/NoReason instead?

Reply:
I would prefer default enum value. I don't think this is enough reason to introduce the complication of Nullable for enum value.

Reply:
ok, used Unknown reason as a default value